lib/sync/types.go

```diff
@@ -0,0 +1,55 @@
+package sync
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"boscoin.io/sebak/lib/ballot"
+	"boscoin.io/sebak/lib/block"
+	"boscoin.io/sebak/lib/transaction"
+)
+
+type SyncProgress struct {
+	StartingBlock uint64 // Block number where sync began
+	CurrentBlock  uint64 // Current block number where sync is at
+	HighestBlock  uint64 // Highest alleged block number in the chain
+}
+
+func (s SyncProgress) Clone() *SyncProgress {
```

Comment:
Since it's a pure value type, we don't really need this, we can just pass by value

Reply:
Hum, I agree that `chan` copies everything. So `Clone()` doesn't need.